tests/transport_test.go

```diff
@@ -86,10 +87,23 @@ var _ = Describe("Transport Tests", func() {
 			Expect(err).NotTo(HaveOccurred())
 			Expect(utils.WaitTimeoutForPodReady(c, sizeCheckPod, ns, 20*time.Second)).To(Succeed())
 
-			command := `expSize=20971520; haveSize=$(wc -c < /pvc/disk.img); (( $expSize == $haveSize )); echo $?`
-			exitCode := f.ExecShellInPod(pod.Name, ns, command)
-			// A 0 exitCode should indicate that $expSize == $haveSize
-			Expect(strconv.Atoi(exitCode)).To(BeZero())
+			switch pvcAnn[controller.AnnSource] {
+			case controller.SourceHTTP:
+				command := `expSize=20971520; haveSize=$(wc -c < /pvc/disk.img); (( $expSize == $haveSize )); echo $?`
+				exitCode := f.ExecShellInPod(pod.Name, ns, command)
+				// A 0 exitCode should indicate that $expSize == $haveSize
+				Expect(strconv.Atoi(exitCode)).To(BeZero())
+			case controller.SourceRegistry:
+				binFile := "/pvc/usr/bin/" + file
+				command := fmt.Sprintf("[ -e %s ]; echo $?", binFile)
+				exitCode := f.ExecShellInPod(pod.Name, ns, command)
+				// A 0 exitCode should indicate that the bin file exists
+				Expect(strconv.Atoi(exitCode)).To(BeZero())
+				if strconv.Atoi(exitCode) != 0 {
```

Comment:
Atoi returns 2 values, so this should be:
```go
if code, _ := strconv.Atoi(exitcode); code != 0 {
...
}
```
Also note that this should be called before the Expect, since the expect will fail and the if statement never executed.